### PR TITLE
Add vitess formula

### PR DESCRIPTION
**Pull request by arthurnn**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
### What is Vitess?
Vitess is a proxy on top of MySQL built by YouTube to add a sharding framework as well as other features on top of MySQL.
See http://vitess.io for more.

### What is in here?
This adds a formula to install vitess binaries.
Note 1: for now, I am pointing to a branch I have, until we figure out how we can vendor the libs into the project. see https://github.com/youtube/vitess/pull/3678
Note 2: Most of Vitess' users build it from Master, that's why I picked the latest SHA there. The latest version is 2.1.1 which is old. Soon we will release version 3. Once that gets released we can bump it here. Is it OK for now to run out of master?

### Which dependencies Vitess need?
It is a go project, so it needs Go to compile it.

cc @sougou
review @MikeMcQuaid

### CODE REVIEW

File: Formula/vitess.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class Vitess < Formula
+  desc "Database clustering system for horizontal scaling of MySQL"
+  homepage "http://vitess.io"
+
+  url "https://github.com/arthurnn/vitess.git",
+      :revision => "75b5eb418f2b5d5ac1e8f5f8fd6bad723d0dcd48",
+      :branch => "arthurnn/easier_build"
+  version "2.1.1.75b5eb41"
+  head "https://github.com/arthurnn/vitess.git"
```
Comment (MikeMcQuaid) [inline Formula/vitess.rb]: Will need to wait until we have an official upstream tag before this gets merged but can review here before that.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/vitess.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Formula/vitess.rb]: Only CI failures relate to that too: https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/19280/version=high_sierra/testReport/junit/brew-test-bot/high_sierra/audit_vitess___online___new_formula/

**Comment by MikeMcQuaid** [comment]
> Is it OK for now to run out of master?

No, unfortunately this will need to be a stable, tagged release before we can include it into this Homebrew repository.

**Comment by fxcoudert** [comment]
> for now, I am pointing to a branch I have, until we figure out how we can vendor the libs into the project

We will need to be able to compile from the main repo, not a fork.

**Pull request closed without merging**